### PR TITLE
Fixes #9186 - build interface if needed during facts import

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by dLobatog** [comment]
@ares Makes sense, can you add a test for a host without a build interface to verify it keeps working?

**Comment by ares** [comment]
@eLobato good catch, I added test, please take another look

**Comment by dLobatog** [comment]
Cool, works well. Merged as ad22248dd01e8192658bd514171520d4d50b1751 , thanks @ares

File: test/unit/host_test.rb [hunk 1]
```diff
@@ -234,6 +234,19 @@ def teardown
...
+    host = FactoryGirl.create(:host, :managed => false)
+    host.interfaces = []
+    host.populate_fields_from_facts(:domain => 'example.com',
+                                    :operatingsystem => 'RedHat',
+                                    :operatingsystemrelease => '6.2',
+                                    :macaddress_eth0 => '00:00:11:22:11:22',
+                                    :ipaddress_eth0 => '192.168.0.1',
+                                    :interfaces => 'eth0')
+    assert_equal 'example.com', host.domain.name
+    refute host.primary_interface.managed?
```
Comment (dLobatog) [inline test/unit/host_test.rb:247]: This is effectively testing that primary_interface was created right?

Comment (ares) [inline test/unit/host_test.rb:247]: yes, maybe a bit better put, that it was built and set as unmanaged

**Pull request closed without merging**